src/main/scala/uk/gov/hmrc/ct/computations/CP88.scala

```diff
@@ -16,11 +16,41 @@
 
 package uk.gov.hmrc.ct.computations
 
-import uk.gov.hmrc.ct.box.{CtBoxIdentifier, CtOptionalInteger, Input}
+import uk.gov.hmrc.ct.box._
+import uk.gov.hmrc.ct.computations.Validators.ComputationValidatableBox
+import uk.gov.hmrc.ct.computations.retriever.ComputationsBoxRetriever
 
-case class CP88(value: Option[Int]) extends CtBoxIdentifier(name = "Annual Investment Allowance") with CtOptionalInteger with Input
+case class CP88(value: Option[Int]) extends CtBoxIdentifier(name = "Annual Investment Allowance") with CtOptionalInteger with Input with ComputationValidatableBox[ComputationsBoxRetriever] {
+  override def validate(boxRetriever: ComputationsBoxRetriever) = {
+    validateZeroOrPositiveInteger(this) ++
+      fieldhasValueWhenTrading(boxRetriever, "CP88", value) ++
+      firstYearAllowanceNotGreaterThanMaxFYA(boxRetriever)
+  }
+
+  private def firstYearAllowanceNotGreaterThanMaxFYA(retriever: ComputationsBoxRetriever): Set[CtValidation] = {
+    val expenditureQualifyingForFirstYearAllowanceInput: Option[Int] = retriever.retrieveCP81Input().value
+    val aiaThreshold: Int = retriever.retrieveCATO02().value
+
+    val maxAIA = Math.min(expenditureQualifyingForFirstYearAllowanceInput.getOrElse(0), aiaThreshold)
+
+    value match {
+      case Some(aiaClaimed) =>
```

Comment:
Your right about this and all the other points above ... however ... I deliberately kept the code as simmilar as possible to original code in front end so that I wouldnt be caught out by some subtle change in behaviour. I like to keep it the way it is for the time being.